src/pages/api/auth-connect/verify-two-fa-validation-code.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  checkVerificationCode,
+  CheckVerificationCodeStatus,
+  ConnectUnreachableError,
+  GraphqlErrors,
+} from "@fewlines/connect-management";
+import {
+  Endpoint,
+  getServerSideCookies,
+  HttpStatus,
+  setAlertMessagesCookie,
+} from "@fwl/web";
+import {
+  loggingMiddleware,
+  wrapMiddlewares,
+  tracingMiddleware,
+  errorMiddleware,
+  recoveryMiddleware,
+  rateLimitingMiddleware,
+} from "@fwl/web/dist/middlewares";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Handler } from "@src/@types/handler";
+import { UserCookie } from "@src/@types/user-cookie";
+import { removeExpiredSudoEventIds } from "@src/commands/remove-expired-sudo-event-ids";
+import { config } from "@src/config";
+import { logger } from "@src/configs/logger";
+import getTracer from "@src/configs/tracer";
+import { NoDBUserFoundError } from "@src/errors/errors";
+import { ERRORS_DATA, webErrorFactory } from "@src/errors/web-errors";
+import { authMiddleware } from "@src/middlewares/auth-middleware";
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { getDBUserFromSub } from "@src/queries/get-db-user-from-sub";
+import { generateAlertMessage } from "@src/utils/generateAlertMessage";
+
+const handler: Handler = async (request, response) => {
+  const webErrors = {
+    identityNotFound: ERRORS_DATA.IDENTITY_NOT_FOUND,
+    sudoEventIdsNotFound: ERRORS_DATA.SUDO_EVENT_IDS_NOT_FOUND,
+    sudoEventIdsListNotFound: ERRORS_DATA.SUDO_EVENT_IDS_LIST_NOT_FOUND,
+    connectUnreachable: ERRORS_DATA.CONNECT_UNREACHABLE,
+    databaseUnreachable: ERRORS_DATA.DATABASE_UNREACHABLE,
+    invalidBody: ERRORS_DATA.INVALID_BODY,
+    invalidValidationCode: ERRORS_DATA.INVALID_VALIDATION_CODE,
+    noUserFound: ERRORS_DATA.NO_USER_FOUND,
+  };
+
+  return getTracer().span(
+    "verify-two-fa-validation-code handler",
+    async (span) => {
+      const { verificationCode } = request.body;
+
+      if (!verificationCode) {
+        throw webErrorFactory(webErrors.invalidBody);
+      }
+
+      const userCookie = (await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: config.cookieSalt,
+      })) as UserCookie;
```

Comment:
Why using a type assertion instead of
```
      if (!userCookie) {
        response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
        response.setHeader("location", "/");
        response.end();
        return;
      }
  ```
  ?

Reply:
Well, I should change it too in `verify-validation-code` then.

Reply:
Hm, I still dunno which one to choose tbh.

Reply:
To uniform all handlers, I think we should put this code snippet in the two verification handlers that had not it yet and then we'll see what we do, what do you think ?